test/CMakeLists.txt

```diff
@@ -161,6 +161,12 @@ target_compile_definitions(ArborX_Clustering.exe PRIVATE BOOST_TEST_DYN_LINK)
 target_include_directories(ArborX_Clustering.exe PRIVATE ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_SOURCE_DIR}/examples/dbscan)
 add_test(NAME ArborX_Clustering_Test COMMAND ./ArborX_Clustering.exe)
 
+add_executable(ArborX_TreeHelpers.exe tstTreeHelpers.cpp utf_main.cpp)
```

Comment:
Placeholder so we can think to a better file/exe name.

Reply:
Yeah, same. I don't know what name should the whole thing be, both the test and the name of the file.

Reply:
What about `TreeNodeLabelling`?

Reply:
Do we envision writing something more general in the future? Like a generic `struct DataUpdater{ void operator()(Data const& left_child_data, Data const& right_child_data)`};`? I'm thinking something like `TreeReducer` could also be ok.